### PR TITLE
Linux 4.7 compat

### PR DESCRIPTION
**Pull request by tuxoko**
Fix handler->get()
Fix blk_queue_write_cache() replace blk_queue_flush()
Fix posix_acl_release() GPL test (not linux 4.7 related)
Fix fops->iterate_shared
Requires-spl:refs/pull/549/head

### CODE REVIEW

File: config/kernel-xattr-handler.m4 [hunk 1]
```diff
@@ -62,74 +62,97 @@ dnl # Supported xattr handler get() interfaces checked newest to oldest.
...
-	dnl # 4.4 API change,
-	dnl # The xattr_handler->get() callback was changed to take a
-	dnl # attr_handler, and handler_flags argument was removed and
-	dnl # should be accessed by handler->flags.
+	dnl # 4.7 API change,
+	dnl # The xattr_handler->get() callback was changed to take both
+	dnl # dentry and inode.
 	dnl #
-	AC_MSG_CHECKING([whether xattr_handler->get() wants xattr_handler])
+	AC_MSG_CHECKING([whether xattr-handler->get() wants both dentry and inode])
```
Comment (behlendorf) [inline config/kernel-xattr-handler.m4]: s/xattr-handler/xattr_handler/

**Comment by behlendorf** [comment]
@tuxoko thanks for tackling this pile of minor compatibility issues.  They really like to change that xattr_handler interface!  This LGTM, just the minor xattr-handler nit.  These checks could definitely be further improved but that shouldn't stop us from adding this support now.

Go ahead and refresh this patch stack and the spl counterpart with a few tiny fixes and we can get it in.

**Comment by tuxoko** [comment]
Fixed the typo.

**Comment by behlendorf** [comment]
LGTM

**Comment by behlendorf** [comment]
@nedbass when cherry-picking this for 0.6.5.8 make sure you also get zfsonlinux/spl@fdbc1ba99d1f4d3958189079eee9b6c957e0264b.  @tuxoko thanks for jumping on this right away.

**Comment by kpande** [comment]
@nedbass any chance of a new release with kernel compatibility and various deadlock fixes from HEAD?

**Comment by stevleibelt** [comment]
@nedbass @behlendorf 

any plan or ideas when this will be released?
Arch linux is using kernel 4.7 already which prevents me from updating a bunch of software :-(.

**Pull request closed without merging**